packages/opentelemetry-grpc-utils/test/grpcUtils.test.ts

```diff
@@ -0,0 +1,690 @@
+                status: grpc.status.OK,
+              };
+
+              assert.strictEqual(spans.length, 2);
+              assertSpan(
+                moduleName,
+                incomingSpan,
+                SpanKind.SERVER,
+                validations
+              );
+              assertSpan(
+                moduleName,
+                outgoingSpan,
+                SpanKind.CLIENT,
+                validations
+              );
+              assertPropagation(incomingSpan, outgoingSpan);
+            } else {
+              assert.strictEqual(spans.length, 0);
+            }
+          });
+      });
+
+      it(`should raise an error for client childSpan/server rootSpan - ${method.description} - status = OK`, () => {
+        const expectEmpty = memoryExporter.getFinishedSpans();
+        assert.strictEqual(expectEmpty.length, 0);
+
+        const span = provider
+          .getTracer('default')
+          .startSpan('TestSpan', { kind: SpanKind.PRODUCER });
+        return provider.getTracer('default').withSpan(span, async () => {
+          const rootSpan = provider.getTracer('default').getCurrentSpan();
+          if (!rootSpan) {
+            return assert.ok(false);
+          }
+          assert.deepStrictEqual(rootSpan, span);
+
+          const args = [client, method.request];
+          await (method.method as any)
+            .apply({}, args)
+            .then(() => {
+              // Assert
+              if (checkSpans) {
+                const spans = memoryExporter.getFinishedSpans();
+                assert.strictEqual(spans.length, 2);
+                const serverSpan = spans[0];
+                const clientSpan = spans[1];
+                const validations = {
+                  name: `grpc.pkg_test.GrpcTester/${method.methodName}`,
+                  status: grpc.status.OK,
+                };
+                assertSpan(
+                  moduleName,
+                  serverSpan,
+                  SpanKind.SERVER,
+                  validations
+                );
+                assertSpan(
+                  moduleName,
+                  clientSpan,
+                  SpanKind.CLIENT,
+                  validations
+                );
+                assertPropagation(serverSpan, clientSpan);
+                assert.strictEqual(
+                  rootSpan.context().traceId,
+                  serverSpan.spanContext.traceId
+                );
+                assert.strictEqual(
+                  rootSpan.context().spanId,
+                  clientSpan.parentSpanId
+                );
+              }
+            })
+            .catch((err: ServiceError) => {
+              assert.ok(false, err);
+            });
+        });
+      });
+    };
+
+    const insertError = (
+      request: TestRequestResponse | TestRequestResponse[]
+    ) => (code: number) =>
+      request instanceof Array
+        ? request.splice(0, 0, { num: code }) &&
+          request.slice(0, request.length)
```

Comment:
```suggestion
        ? [{ num: code }, ...request]
```

In this way you can get rid of the copy logic at the call site, as this function is no longer modifying the input.